From a5da0766fd5e0a9ce5ac9e033cbeda1058e8a588 Mon Sep 17 00:00:00 2001
From: Gary Liu <garyliu@rbcon.com>
Date: Thu, 13 Feb 2014 13:18:04 +0800
Subject: [PATCH] replace SNAT with MASQ

---
 warden/root/linux/net.sh | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/warden/root/linux/net.sh b/warden/root/linux/net.sh
index 423dc41..c2c8ad7 100755
--- a/warden/root/linux/net.sh
+++ b/warden/root/linux/net.sh
@@ -163,11 +163,10 @@ function setup_nat() {
       --jump ${nat_postrouting_chain}
 
   # Enable NAT for traffic coming from containers
-  (iptables -t nat -S ${nat_postrouting_chain} | grep -q "\-j SNAT\b") ||
+  (iptables -t nat -S ${nat_postrouting_chain} | grep -q "\-j MASQUERADE\b") ||
     iptables -t nat -A ${nat_postrouting_chain} \
       --source ${POOL_NETWORK} \
-      --jump SNAT \
-      --to $(external_ip)
+      --jump MASQUERADE
 }
 
 case "${1}" in
-- 
1.8.3.4 (Apple Git-47)

